src/test/java/uk/gov/companieshouse/controller/DissolutionControllerTest.java

```diff
@@ -198,38 +201,66 @@ public void submitDissolutionRequest_returnsInternalServerError_ifExceptionOccur
         when(service.create(isA(DissolutionCreateRequest.class), eq(COMPANY_NUMBER), eq(USER_ID), eq(IP_ADDRESS), eq(EMAIL))).thenThrow(new RuntimeException());
 
         mockMvc
-            .perform(
-                post(DISSOLUTION_URI, COMPANY_NUMBER)
-                    .contentType(MediaType.APPLICATION_JSON_VALUE)
-                    .headers(createHttpHeaders())
-                    .content(asJsonString(body)))
-            .andExpect(status().isInternalServerError());
+                .perform(
+                        post(DISSOLUTION_URI, COMPANY_NUMBER)
+                                .contentType(MediaType.APPLICATION_JSON_VALUE)
+                                .headers(createHttpHeaders())
+                                .content(asJsonString(body)))
+                .andExpect(status().isInternalServerError());
 
         verify(service).create(isA(DissolutionCreateRequest.class), eq(COMPANY_NUMBER), eq(USER_ID), eq(IP_ADDRESS), eq(EMAIL));
     }
 
+    @Test
+    public void getDissolutionRequest_returnsUnauthorised_ifEricIdentityHeaderIsBlank() throws Exception {
+        final HttpHeaders headers = new HttpHeaders() {{
+            add(IDENTITY_HEADER, "");
+            add(AUTHORISED_USER_HEADER, EMAIL);
+        }};
+
+        mockMvc
+                .perform(
+                        get(DISSOLUTION_URI, COMPANY_NUMBER)
+                                .contentType(MediaType.APPLICATION_JSON_VALUE)
+                                .headers(headers)
+                )
+                .andExpect(status().isUnauthorized());
+    }
+
+    @Test
+    public void getDissolutionRequest_returnsNotFound_ifDissolutionAlreadyExistsForCompany() throws Exception {
```

Comment:
this test name seems wrong.

Also, we are missing a test for what happens if a Dissolution is found for a company.